mathics/builtin/lists.py

```diff
@@ -957,18 +957,22 @@ def apply_range(self, expr, i, imax, evaluation):
     def apply_max(self, expr, imax, evaluation):
         "%(name)s[expr_, {imax_}]"
 
-        index = 0
-        imax = imax.evaluate(evaluation)
-        imax = imax.numerify(evaluation)
-        if isinstance(imax, Number):
-            imax = imax.round()
-        imax = imax.get_float_value()
-        if imax is None:
-            if self.throw_iterb:
-                evaluation.message(self.get_name(), "iterb")
-            return
+        if isinstance(imax, Integer):
```

Comment:
Nop: In WMA, 
```
In[2]:= Do[Print[i],{2.3}]                                                      
i
i
```
So, imax could be perfectly a Real number.

Reply:
In any case, I agree with the proposed change.

Reply:
Ok - Thanks for the information. 'll make note of this in a comment somewhere.